src/assets/asset_manager.rs

```diff
@@ -8,15 +8,15 @@ use walkdir::WalkDir;
 use crate::core::Font;
 use crate::graphics::{
     material::{Image, NewMaterialData, NewMaterialHandle, Shader},
-    mesh::{GltfData, SubMesh},
+    mesh::{GltfData, SubMesh, Mesh},
     resources::GPUResourceManager,
 };
 
 pub struct AssetManager {
     path: String,
     shaders: HashMap<String, Shader>,
     fonts: HashMap<String, Font>,
-    meshes: HashMap<String, SubMesh>,
+    meshes: HashMap<String, Mesh>,
     pub(crate) materials: HashMap<NewMaterialHandle, Option<Arc<NewMaterialData>>>,
```

Comment:
> Does this link to the list of materials by ID?

@StarArawn 
Materials can be identified by their hashed materialhandle. Hashcollisions should not appear. This makes sure that Materials wont get loaded multiple times. We can do hot reload without unloading or reloading materials like this.